packages/component-test-utils-react/src/dispatcher/index.spec.js

```diff
@@ -35,4 +35,40 @@ describe('dispatcher', () => {
       expect(finalState).toBe(4);
     });
   });
+
+  describe('useEffect', () => {
+    it('should call useEffect on firstRender', () => {
+      const useEffectFn = jest.fn();
+      dispatcher.useEffect(useEffectFn);
+
+      expect(useEffectFn).toHaveBeenCalledTimes(1);
+    });
+
+    it('should call useEffect each time a render apear when non memoization is set', () => {
+      const useEffectFn = jest.fn();
+      dispatcher.useEffect(useEffectFn);
+      dispatcher._informDipatcherRenderIsDone();
+      dispatcher.useEffect(useEffectFn);
+      dispatcher._informDipatcherRenderIsDone();
+
+      expect(useEffectFn).toHaveBeenCalledTimes(2);
+      dispatcher.useEffect(useEffectFn);
+      expect(useEffectFn).toHaveBeenCalledTimes(3);
+    });
+
+    it('should call useEffect when a render apear and when props changes', () => {
```

Comment:
You've read some talks of Mancuso to not rely on `if` but `when` in your test cases? 😛

Reply:
not at all 😅 

Any link ?

Reply:
I ll try to find :p